pkg/logs/decoder/decoder.go

```diff
@@ -72,6 +72,14 @@ func InitializeDecoder(source *config.LogSource, parser parser.Parser) *Decoder
 	return New(inputChan, outputChan, lineHandler, lineLimit)
 }
 
+// InitializeDecoderWithDockerParser returns a properly initialized Decoder with docker parser
+// TODO: Refactor decoder to be more specific per source
+func InitializeDecoderWithDockerParser(source *config.LogSource, parser parser.Parser) *Decoder {
```

Comment:
Why adding the specific logic to the common decoder ? I'd rather duplicate some code and isolate the new logic ; why ? Because if we broke the decoder, it will only be broken for docker logs, plus we enter the `matchDockerHeader` method and do the sanity `useDockerParser` all the time when it's not needed.

Reply:
What about moving the specific decoder to the docker package as well ?

Reply:
I am not sure replicating the majority of the code for specific source is better. Both way we are hacking this decoder. I would want to keep the code in the same place.

Reply:
The flag `useDockerParser` is assigned when initializing decoder, I don't think it will effect performance to do this check.